smack-integration-test/src/main/java/org/jivesoftware/smackx/chat/IncomingMessageListenerIntegrationTest.java

```diff
@@ -0,0 +1,64 @@
+/**
+ *
+ * Copyright 2018 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.chat;
+
+import org.jivesoftware.smack.chat2.Chat;
+import org.jivesoftware.smack.chat2.ChatManager;
+import org.jivesoftware.smack.chat2.IncomingChatMessageListener;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.util.StringUtils;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.igniterealtime.smack.inttest.util.SimpleResultSyncPoint;
+import org.junit.After;
+import org.jxmpp.jid.EntityBareJid;
+
+public class IncomingMessageListenerIntegrationTest extends AbstractSmackIntegrationTest {
+
+    public IncomingMessageListenerIntegrationTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+    }
+
+    private final String body = StringUtils.randomString(16);
+    private final SimpleResultSyncPoint syncPoint = new SimpleResultSyncPoint();
+    private final IncomingChatMessageListener listener = new IncomingChatMessageListener() {
+        @Override
+        public void newIncomingMessage(EntityBareJid from, Message message, Chat chat) {
+            if (body.equals(message.getBody())) {
+                syncPoint.signal();
+            }
+        }
+    };
+
+    @SmackIntegrationTest
+    public void test() throws Exception {
+        ChatManager m1 = ChatManager.getInstanceFor(conOne);
```

Comment:
The ChatManagers should be fields of the test class. And they should be named after the patter the other tests use, i.e. `chatManagerOne` and `chatManagerTwo`.